activerecord/test/cases/nested_attributes_test.rb

```diff
@@ -1206,12 +1206,19 @@ def self.name; "Guitar"; end
 end
 
 class TestNestedAttributesWithExtend < ActiveRecord::TestCase
-  setup do
-    Pirate.accepts_nested_attributes_for :treasures
-  end
-
   def test_extend_affects_nested_attributes
-    pirate = Pirate.create!(catchphrase: "Don' botharrr talkin' like one, savvy?")
+    super_pirate = Class.new(ActiveRecord::Base) do
```

Comment:
Unrelated to this change, but `Pirate.accepts_nested_attributes_for` leaks to other tests. (It makes asserting errors message from relation different as it adds `autosave: true` to the reflection).